cmd/signature-v4_test.go

```diff
@@ -54,14 +54,7 @@ func TestDoesPolicySignatureMatch(t *testing.T) {
 			},
 			expected: ErrInvalidAccessKeyID,
 		},
-		// (2) It should fail if the region is invalid.
-		{
-			form: http.Header{
-				"X-Amz-Credential": []string{fmt.Sprintf(credentialTemplate, accessKey, now.Format(yyyymmdd), "invalidregion")},
```

Comment:
Yes but the case i am talking about is when server has a region set and a malformed X-Amz-Credential is sent, we should send ErrInvalidRegion in that scenario. Such a case is not covered here. IMHO.